### PR TITLE
add missing center to grid coordinates

### LINKED ISSUE #4: Possible 2d rotation error
* torch-rotate-image version: N/A
* Python version: 3.10.14
* Operating System: macOS 14.5

### Description

`test_rotate_image_2d_rotation()` is failing because it appears that that is an error with rotation. E.g. I would expect that applying a 180° rotation would produce a flipped image but it does not. This may be an error in my interpretation of the code, I need to get to the bottom of it still. My current guess is that it has to do with the `torch_image_lerp.sample_image_2d` call in `rotate_image_2d`.

### What I Did

This is from the contents of `tests.test_rotate_image_2d_rotation`:
```python
image = torch.zeros(28, 28)
image[:14, :] = 1  # image is half black, half white
angles = torch.tensor([0.0, 90.0, 180.0])
rotated_image = rotate_image_2d(image, angles) # (3, 28, 28)
```

`image`
image

`rotated_image[0]`
image

`rotated_image[1]`
image

`rotated_image[2]`
image

### PR DESCRIPTION
**Pull request by alisterburt**
closes #4 

@sjrothfuss we had centered coordinates and rotated them but we need to un-center them for sampling. This is needed because the sampling utilities in torch image lerp expect coordinates which align with array indices `[0, n-1]` for each dim

hope this helps!

### CODE REVIEW

**Comment by alisterburt** [comment]
I'm unsure if the test will pass here, the rotation center being on the DC component might bring a strip of out of bounds pixels into the result

**Comment by alisterburt** [comment]
will need to coerce the center to tensor with the right dtype and on the right device - take this as a push in the right direction rather than a complete solution, good luck!
